src/client/src/ui/spotTile/components/SpotTile.tsx

```diff
@@ -38,7 +39,22 @@ export default class SpotTile extends PureComponent<Props, State> {
 
   updateNotional = (notional: string) => this.setState({ notional })
 
-  executeTrade = (direction: Direction) => this.props.executeTrade(direction, numeral(this.state.notional).value())
+  createTrade = (direction: Direction) => {
+    const { spotTileData, currencyPair, executeTrade } = this.props
+    if (spotTileData.isTradeExecutionInFlight || !spotTileData.price) {
```

Comment:
You could also check if execution is disabled here. 

Or we can just disable the trading button when `spotTileData.isTradeExecutionInFlight || !spotTileData.price` is true and remove the code from here.